src/queries/hintText.ts

```diff
@@ -10,16 +9,27 @@ import type {
   QueryAllByQuery,
   QueryByQuery,
 } from './makeQueries';
+import { TextMatchOptions } from './text';
+
+const getNodeByHintText = (
+  node: ReactTestInstance,
+  text: TextMatch,
+  options: TextMatchOptions = {}
+) => {
+  const { exact, normalizer } = options;
+  return matches(text, node.props.accessibilityHint, normalizer, exact);
+};
 
 const queryAllByHintText = (
   instance: ReactTestInstance
-): ((hint: TextMatch) => Array<ReactTestInstance>) =>
-  function queryAllByA11yHintFn(hint) {
-    return instance.findAll(
-      (node) =>
-        typeof node.type === 'string' &&
-        matchStringProp(node.props.accessibilityHint, hint)
-    );
+): ((
+  hint: TextMatch,
+  queryOptions?: TextMatchOptions
+) => Array<ReactTestInstance>) =>
+  function queryAllByA11yHintFn(hint, queryOptions) {
+    return instance
+      .findAll((node) => getNodeByHintText(node, hint, queryOptions))
+      .filter((element) => typeof element.type === 'string');
```

Comment:
Please move this check back in the findAll and before the `getNodeByHintText`. Checking the `type === 'string'` is the cheapest check, and we try to run the cheapest checks first.

Reply:
Have added the same in latest